webknossos-datastore/app/com/scalableminds/webknossos/datastore/datareaders/Compressor.scala

```diff
@@ -268,16 +295,39 @@ class JpegCompressor() extends Compressor {
   override def toString: String = getId
 
   @throws[IOException]
-  override def compress(is: InputStream, os: OutputStream): Unit = ???
+  override def compress(input: Array[Byte]): Array[Byte] = ???
 
   @throws[IOException]
-  override def uncompress(is: InputStream, os: OutputStream): Unit = {
+  override def decompress(input: Array[Byte]): Array[Byte] = {
+    val is = new ByteArrayInputStream(input)
     val iis: ImageInputStream = createImageInputStream(is)
     val bi: BufferedImage = ImageIO.read(iis: ImageInputStream)
     val raster = bi.getRaster
     val dbb: DataBufferByte = raster.getDataBuffer.asInstanceOf[DataBufferByte]
     val width = raster.getWidth
     val data = dbb.getData.grouped(width).toList
-    os.write(data.flatten.toArray)
+    data.flatten.toArray
   }
 }
+
+class CompressedSegmentationCompressor(dataType: PrecomputedDataType, volumeSize: Array[Int], blockSize: Array[Int])
+    extends Compressor {
+  override def getId: String = "compressedsegmentation"
+
+  override def toString: String = s"compressor=$getId/dataType=${dataType.toString}"
+
+  private def blockSizeVec = Vec3Int(x = blockSize(0), y = blockSize(1), z = blockSize(2))
```

Comment:
could blockSize be Vec3Int everywhere? (even in case class PrecomputedScale). Then the json validator would already reject it if it does not match, and we wouldn’t need the conversion here

Reply:
oh I did not know Vec3Int could be automatically parsed.